### ISSUE TITLE
Fluid-Level XDRs are ignored when no other Engine XDRs are received

### ISSUE THREAD

**Issue by ch-fb** [TwoCanPlugIn/EngineDashboard#4]
Hi again,
Either I'm doing something completely wrong, or there is a bug: If the input only has XDRs for Tanks, the tank data is not displayed. As soon as an RPM sentence is received, the tank info is shown. As soon as RMP sentence stops again, tank display stops as well.
with best regards,
Christian

**Comment by TwoCanPlugIn** [comment]
No, you are not doing anything wrong. There is a watchdog timer that automatically zeros the gauges if no data, and in particular no RPM's have been received within a certain time limit. Imagine the scenario, the engine is switched off. The oil pressure and alternator potential would be zero. So I need to have some way of determining when to zero the gauges. Unfortunately as you have discovered, it means you cannot observe fluid levels if the engine is not running. A solution maybe is that I ignore the watchdog timer for fluid levels.

**Comment by ch-fb** [comment]
i thought so... you could also add another timer for fluidlevels

**Comment by TwoCanPlugIn** [comment]
Yes, that is a possible solution.

BTW, What multiplexer/gateway and engine/tank level sensors are you using ?

Also what OS are you using OpenCPN & the Engine dashboard on. I may be able to implement some alpha/beta releases for you to test.

**Comment by TwoCanPlugIn** [comment]
Fortunately the order within the code for the different gauge instances made for a simple fix that leaves the tank level & battery voltage gauges still displaying values while he engine is off. Will be in the next release which is going through the automated build process, so I shall now close this.
Additional tank level gauges, is not necessarily difficult, just a bit more coding which I may be able to do in a later release.

**Comment by ch-fb** [comment]
Excellent, thanks a lot!
cheers,
C
